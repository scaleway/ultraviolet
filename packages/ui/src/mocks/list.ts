export type MockData = {
  department: string
  description: string
  id: string
  name: string
  reference: string | number
}

export const generateData = (count?: number, prefix?: string): MockData[] =>
  new Array(count).map((_, index) => ({
    department: index % 3 === 0 ? `Front` : `Not Front`,
    description: `Fake message for row ${index}`,
    id: `${prefix ? `${prefix}-` : ''}${index}`,
    name: `Scaler ${prefix ? `${prefix}-` : ''}${index}`,
    reference: prefix ? `${prefix}-${index}` : index,
  }))

const listMock = { generateData }

export default listMock
