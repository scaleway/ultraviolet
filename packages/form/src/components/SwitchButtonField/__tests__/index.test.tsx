import { renderHook, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm, renderWithTheme } from '@utils/test'
import { useForm } from 'react-hook-form'
import { describe, expect, test, vi } from 'vitest'
import { mockErrors } from '../../../mocks'
import { Form, Submit, SwitchButtonField } from '../..'

describe('SwitchButtonField', () => {
  test('should render correctly', () => {
    const { asFragment } = renderWithForm(
      <SwitchButtonField name="test" onChange={() => vi.fn()}>
        <SwitchButtonField.Option value="left">Left</SwitchButtonField.Option>
        <SwitchButtonField.Option value="right">Right</SwitchButtonField.Option>
      </SwitchButtonField>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should works with defaultValues', async () => {
    const onSubmit = vi.fn()
    const { result } = renderHook(() =>
      useForm<{ test: string[] }>({ defaultValues: { test: ['right'] } }),
    )

    const { asFragment } = renderWithTheme(
      <Form onSubmit={onSubmit} errors={mockErrors} methods={result.current}>
        <SwitchButtonField name="test" onChange={() => vi.fn()}>
          <SwitchButtonField.Option value="left">Left</SwitchButtonField.Option>
          <SwitchButtonField.Option value="right">
            Right
          </SwitchButtonField.Option>
        </SwitchButtonField>
        ,<Submit>Submit</Submit>
      </Form>,
    )

    await userEvent.click(screen.getByText('Submit'))
    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalledTimes(1)
    })
    expect(onSubmit.mock.calls[0][0]).toEqual({
      test: ['right'],
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
