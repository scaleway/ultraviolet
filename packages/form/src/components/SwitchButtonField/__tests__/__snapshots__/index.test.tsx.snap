// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SwitchButtonField > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  position: relative;
}

.emotion-0[data-size='small']>.emotion-2 {
  height: 2.5rem;
}

.emotion-0[data-size='medium']>.emotion-2 {
  height: 3rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
  border: none;
  padding: 0.5rem 1rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  white-space: nowrap;
  background: transparent;
}

.emotion-3[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-3[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-3[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-3[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-3[data-image="illustration"] {
  padding: 0rem;
}

.emotion-3[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-3:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-3:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-3:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-3:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-3[data-has-label='true'] .emotion-6,
.emotion-3[data-has-label='true'] .eqr7bqq1 {
  width: 100%;
}

.emotion-3:hover,
.emotion-3:active {
  box-shadow: none;
  border: none;
}

.emotion-3:hover:not([data-error='true'][data-disabled='true']),
.emotion-3:active:not([data-error='true'][data-disabled='true']) {
  border: none;
}

.emotion-3[data-checked='true'] {
  border: none;
}

.emotion-3[data-checked='true'] label {
  color: #ffffff;
}

.emotion-3:not([data-checked='true']) label:hover {
  color: #641cb3;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-8[aria-disabled='false'],
.emotion-8[aria-disabled='false']>label {
  cursor: pointer;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 {
  fill: #8c40ef;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 .emotion-15 {
  fill: #e5dbfd;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffd3e3;
}

.emotion-8[aria-disabled='true'] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-8[aria-disabled='true']>label,
.emotion-8[aria-disabled='true'] .emotion-11 {
  cursor: not-allowed;
}

.emotion-8[aria-disabled='true'] .emotion-13 {
  fill: #e9eaeb;
  cursor: not-allowed;
}

.emotion-8[aria-disabled='true'] .emotion-13 .emotion-15 {
  fill: #f3f3f4;
}

.emotion-8[data-checked='true'] {
  color: #641cb3;
}

.emotion-8[data-error='true'] {
  color: #b3144d;
}

.emotion-8[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-8 input+svg {
  display: none;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11+.emotion-13 {
  fill: #d9dadd;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 {
  fill: #8c40ef;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8 .emotion-11[aria-disabled='false']:active+.emotion-13 {
  background: none;
  fill: #8c40ef;
}

.emotion-8 .emotion-11[aria-disabled='false']:active+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-10 {
  cursor: pointer;
  position: absolute;
  height: 1.5rem;
  width: 1.5rem;
  opacity: 0;
  white-space: nowrap;
  border-width: 0;
}

.emotion-10+.emotion-13 .emotion-17 {
  transform-origin: center;
  -webkit-transition: 200ms -webkit-transform ease-in-out;
  transition: 200ms transform ease-in-out;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.emotion-10:checked+svg .emotion-17 {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-10:checked[aria-disabled='false'][aria-invalid='false']+.emotion-13 {
  fill: #8c40ef;
}

.emotion-10:checked[aria-disabled='true'][aria-invalid='false']+.emotion-13 {
  fill: #d8c5fc;
}

.emotion-10[aria-invalid='true']:not([aria-disabled='true'])+.emotion-13 {
  fill: #e51963;
}

.emotion-10[aria-disabled='false']:active+.emotion-13 {
  background-color: #5e127e40;
  fill: #8c40ef;
}

.emotion-10[aria-disabled='false']:active+.emotion-13 .emotion-15 {
  fill: #f1eefc;
}

.emotion-10[aria-disabled='false']:focus-visible+.emotion-13 {
  outline: -webkit-focus-ring-color auto 1px;
}

.emotion-10[aria-invalid='true']:focus+.emotion-13 {
  background-color: #f91b6c40;
  fill: #e51963;
}

.emotion-10[aria-invalid='true']:focus+.emotion-13 .emotion-15 {
  fill: #ffebf2;
}

.emotion-12 {
  height: 1.5rem;
  width: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  border-radius: 100%;
  fill: #d9dadd;
}

.emotion-12 .emotion-15 {
  fill: #ffffff;
}

.emotion-19 {
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
    >
      <div
        style="display: inline-flex;"
      >
        <div
          class="emotion-0 emotion-1"
          data-size="small"
        >
          <div
            class="emotion-2 emotion-3 emotion-4"
            data-checked="false"
            data-disabled="false"
            data-has-label="true"
            data-image="none"
            data-type="radio"
            role="button"
            tabindex="0"
          >
            <div
              class="emotion-5 emotion-6"
            >
              <div
                aria-disabled="false"
                class="emotion-7 emotion-8 emotion-9"
                data-checked="false"
              >
                <input
                  aria-disabled="false"
                  aria-invalid="false"
                  class="emotion-10 emotion-11"
                  id=":r2:"
                  name="test"
                  tabindex="-1"
                  type="radio"
                  value="left"
                />
                <svg
                  class="emotion-12 emotion-13"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                      stroke-width="2"
                    />
                    <circle
                      class="emotion-14 emotion-15"
                      cx="12"
                      cy="12"
                      r="8"
                    />
                    <circle
                      class="emotion-16 emotion-17"
                      cx="12"
                      cy="12"
                      r="5"
                    />
                  </g>
                </svg>
                <label
                  class="emotion-18 emotion-19 emotion-20"
                  for=":r2:"
                >
                  Left Button Label
                </label>
              </div>
            </div>
          </div>
          <div
            class="emotion-2 emotion-3 emotion-4"
            data-checked="true"
            data-disabled="false"
            data-has-label="true"
            data-image="none"
            data-type="radio"
            role="button"
            tabindex="0"
          >
            <div
              class="emotion-5 emotion-6"
            >
              <div
                aria-disabled="false"
                class="emotion-7 emotion-8 emotion-9"
                data-checked="true"
              >
                <input
                  aria-disabled="false"
                  aria-invalid="false"
                  checked=""
                  class="emotion-10 emotion-11"
                  id=":r6:"
                  name="test"
                  tabindex="-1"
                  type="radio"
                  value="right"
                />
                <svg
                  class="emotion-12 emotion-13"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                      stroke-width="2"
                    />
                    <circle
                      class="emotion-14 emotion-15"
                      cx="12"
                      cy="12"
                      r="8"
                    />
                    <circle
                      class="emotion-16 emotion-17"
                      cx="12"
                      cy="12"
                      r="5"
                    />
                  </g>
                </svg>
                <label
                  class="emotion-18 emotion-19 emotion-20"
                  for=":r6:"
                >
                  Right Button Label
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`SwitchButtonField > should works with defaultValues 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
  padding: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  position: relative;
}

.emotion-0[data-size='small']>.emotion-2 {
  height: 2.5rem;
}

.emotion-0[data-size='medium']>.emotion-2 {
  height: 3rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
  border: none;
  padding: 0.5rem 1rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  white-space: nowrap;
  background: transparent;
}

.emotion-3[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-3[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-3[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-3[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-3[data-image="illustration"] {
  padding: 0rem;
}

.emotion-3[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-3:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-3:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-3:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-3:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-3[data-has-label='true'] .emotion-6,
.emotion-3[data-has-label='true'] .eqr7bqq1 {
  width: 100%;
}

.emotion-3:hover,
.emotion-3:active {
  box-shadow: none;
  border: none;
}

.emotion-3:hover:not([data-error='true'][data-disabled='true']),
.emotion-3:active:not([data-error='true'][data-disabled='true']) {
  border: none;
}

.emotion-3[data-checked='true'] {
  border: none;
}

.emotion-3[data-checked='true'] label {
  color: #ffffff;
}

.emotion-3:not([data-checked='true']) label:hover {
  color: #641cb3;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-8[aria-disabled='false'],
.emotion-8[aria-disabled='false']>label {
  cursor: pointer;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 {
  fill: #8c40ef;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 .emotion-15 {
  fill: #e5dbfd;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffd3e3;
}

.emotion-8[aria-disabled='true'] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-8[aria-disabled='true']>label,
.emotion-8[aria-disabled='true'] .emotion-11 {
  cursor: not-allowed;
}

.emotion-8[aria-disabled='true'] .emotion-13 {
  fill: #e9eaeb;
  cursor: not-allowed;
}

.emotion-8[aria-disabled='true'] .emotion-13 .emotion-15 {
  fill: #f3f3f4;
}

.emotion-8[data-checked='true'] {
  color: #641cb3;
}

.emotion-8[data-error='true'] {
  color: #b3144d;
}

.emotion-8[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-8 input+svg {
  display: none;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11+.emotion-13 {
  fill: #d9dadd;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false']:not([data-checked='true']) .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 {
  fill: #8c40ef;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 {
  fill: #b3144d;
}

.emotion-8:hover[aria-disabled='false'] .emotion-11[aria-invalid='true']+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-8 .emotion-11[aria-disabled='false']:active+.emotion-13 {
  background: none;
  fill: #8c40ef;
}

.emotion-8 .emotion-11[aria-disabled='false']:active+.emotion-13 .emotion-15 {
  fill: #ffffff;
}

.emotion-10 {
  cursor: pointer;
  position: absolute;
  height: 1.5rem;
  width: 1.5rem;
  opacity: 0;
  white-space: nowrap;
  border-width: 0;
}

.emotion-10+.emotion-13 .emotion-17 {
  transform-origin: center;
  -webkit-transition: 200ms -webkit-transform ease-in-out;
  transition: 200ms transform ease-in-out;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.emotion-10:checked+svg .emotion-17 {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-10:checked[aria-disabled='false'][aria-invalid='false']+.emotion-13 {
  fill: #8c40ef;
}

.emotion-10:checked[aria-disabled='true'][aria-invalid='false']+.emotion-13 {
  fill: #d8c5fc;
}

.emotion-10[aria-invalid='true']:not([aria-disabled='true'])+.emotion-13 {
  fill: #e51963;
}

.emotion-10[aria-disabled='false']:active+.emotion-13 {
  background-color: #5e127e40;
  fill: #8c40ef;
}

.emotion-10[aria-disabled='false']:active+.emotion-13 .emotion-15 {
  fill: #f1eefc;
}

.emotion-10[aria-disabled='false']:focus-visible+.emotion-13 {
  outline: -webkit-focus-ring-color auto 1px;
}

.emotion-10[aria-invalid='true']:focus+.emotion-13 {
  background-color: #f91b6c40;
  fill: #e51963;
}

.emotion-10[aria-invalid='true']:focus+.emotion-13 .emotion-15 {
  fill: #ffebf2;
}

.emotion-12 {
  height: 1.5rem;
  width: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  border-radius: 100%;
  fill: #d9dadd;
}

.emotion-12 .emotion-15 {
  fill: #ffffff;
}

.emotion-19 {
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

.emotion-40 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3rem;
  padding: 0 1rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  paragraph-spacing: 0;
  text-case: none;
  background: #8c40ef;
  border: none;
  color: #ffffff;
}

.emotion-40:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-40:active {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-40:hover,
.emotion-40:active {
  background: #792dd4;
  color: #f9f9fa;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
    >
      <div
        style="display: inline-flex;"
      >
        <div
          class="emotion-0 emotion-1"
          data-size="small"
        >
          <div
            class="emotion-2 emotion-3 emotion-4"
            data-checked="false"
            data-disabled="false"
            data-has-label="true"
            data-image="none"
            data-type="radio"
            role="button"
            tabindex="0"
          >
            <div
              class="emotion-5 emotion-6"
            >
              <div
                aria-disabled="false"
                class="emotion-7 emotion-8 emotion-9"
                data-checked="false"
              >
                <input
                  aria-disabled="false"
                  aria-invalid="false"
                  class="emotion-10 emotion-11"
                  id=":rb:"
                  name="test"
                  tabindex="-1"
                  type="radio"
                  value="left"
                />
                <svg
                  class="emotion-12 emotion-13"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                      stroke-width="2"
                    />
                    <circle
                      class="emotion-14 emotion-15"
                      cx="12"
                      cy="12"
                      r="8"
                    />
                    <circle
                      class="emotion-16 emotion-17"
                      cx="12"
                      cy="12"
                      r="5"
                    />
                  </g>
                </svg>
                <label
                  class="emotion-18 emotion-19 emotion-20"
                  for=":rb:"
                >
                  Left Button Label
                </label>
              </div>
            </div>
          </div>
          <div
            class="emotion-2 emotion-3 emotion-4"
            data-checked="true"
            data-disabled="false"
            data-has-label="true"
            data-image="none"
            data-type="radio"
            role="button"
            tabindex="0"
          >
            <div
              class="emotion-5 emotion-6"
            >
              <div
                aria-disabled="false"
                class="emotion-7 emotion-8 emotion-9"
                data-checked="true"
              >
                <input
                  aria-disabled="false"
                  aria-invalid="false"
                  checked=""
                  class="emotion-10 emotion-11"
                  id=":rf:"
                  name="test"
                  tabindex="-1"
                  type="radio"
                  value="right"
                />
                <svg
                  class="emotion-12 emotion-13"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                      stroke-width="2"
                    />
                    <circle
                      class="emotion-14 emotion-15"
                      cx="12"
                      cy="12"
                      r="8"
                    />
                    <circle
                      class="emotion-16 emotion-17"
                      cx="12"
                      cy="12"
                      r="5"
                    />
                  </g>
                </svg>
                <label
                  class="emotion-18 emotion-19 emotion-20"
                  for=":rf:"
                >
                  Right Button Label
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="emotion-40 emotion-41"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
